ra2ce/graph/hazard.py

```diff
@@ -887,88 +924,112 @@ def create(self):
                 logging.info(f"Saved {ods_path.stem} in {ods_path.resolve().parent}.")
 
         #### Step 3: iterate overlay of the GeoPandas Dataframe (if any) ###
-        if self.files["base_network"] and self.files["base_network_hazard"] is None:
-            # Check if the graph needs to be reprojected
-            hazard_crs = pyproj.CRS.from_user_input(self.config["hazard"]["hazard_crs"])
-            gdf_crs = pyproj.CRS.from_user_input(self.graphs["base_network"].crs)
+        if self.files["base_network"]:
```

Comment:
take this opportunity to extract this logic.

Reply:
Could you explain what do you mean by "extract this logic"?